src/server/common.h

```diff
@@ -355,4 +355,55 @@ template <typename Mutex> class ABSL_SCOPED_LOCKABLE SharedLock {
 
 extern size_t serialization_max_chunk_size;
 
+struct ConditionFlag {
+  util::fb2::CondVarAny cond_var;
+  bool flag = false;
+};
+
+// Helper class used to guarantee atomicity between serialization of buckets
+class ConditionGuard {
```

Comment:
How is this not just a thread local mutex? 🤔 Topological proof 🤓: you can move the constructor/destructor code into lock and unlock() of ConditionFlag and use it with lock_guard

Reply:
:100: